base/markdown/Common/inline.jl

```diff
@@ -30,14 +30,20 @@ end
 
 @trigger '`' ->
 function inline_code(stream::IO, md::MD)
-    result = parse_inline_wrapper(stream, "`"; rep=true)
-    return result === nothing ? nothing : Code(result)
-end
-
-@trigger '`' ->
-function inline_tex(stream::IO, md::MD)
-    result = parse_inline_wrapper(stream, "``"; rep=true)
-    return result === nothing ? nothing : LaTeX(result)
+    withstream(stream) do
+        ticks = startswith(stream, r"^(`+)")
+        result = readuntil(stream, ticks)
```

Comment:
The `startswith` and `readuntil` aren't from `Base`, they're different functions specific to the `Base.Markdown` module. It's definitely confusing and their names should probably be changed when #12740 gets done.

`Markdown.startswith` returns either `nothing` or the regex match, in this case the string of opening backticks. `Markdown.readuntil` then reads up until the matching closing backticks. If the `do` block returns `nothing` then `withstream` resets `stream` to there it was when entering the block.